internal/parse/parser.go

```diff
@@ -111,15 +112,16 @@ func (p *Parser) add(part queryPart) {
 }
 
 // Parse takes an input string and parses the input and output parts. It returns
-// a pointer to a ParsedExpr.
+// a pointer to a ParsedExpr. If the parser encounters an error then ParsedExpr
```

Comment:
The change here also violates that rule: it's a standard practice in Go to return zero values every time there's an error. We don't need to add that information to every comment.

Reply:
Removed part of comment about error.